mechafil_jax/locking.py

```diff
@@ -139,20 +155,68 @@ def compute_new_pledge_for_added_power(
     total_qa_power_pib: float,
     baseline_power_pib: float,
     lock_target: float,
+    gamma: float,
+    gamma_weight_type: int,  
```

Comment:
This should be a Python enum so that it is more readable.

https://docs.python.org/3/library/enum.html